jni/tests/faiss_stream_support_test.cpp

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+//
+// Modifications Copyright OpenSearch Contributors. See
+// GitHub history for details.
+
+#include "faiss_stream_support.h"
+#include <gmock/gmock.h>
+#include "test_util.h"
+#include <jni.h>
+#include <random>
+#include <algorithm>
+#include <vector>
+
+using ::testing::Return;
+using knn_jni::stream::FaissOpenSearchIOReader;
+using knn_jni::stream::NativeEngineIndexInputMediator;
+using test_util::MockJNIUtil;
+
+// Mocking IndexInputWithBuffer.
+struct JavaIndexInputMock {
```

Comment:
Seems like copy pasta, any possibility to make it reusable?

Reply:
em... which part? 🤔  
Probably `ON_CALL` parts look promising to be shared across though, but not sure they are..
They are there for reasons for mocking method behavior (`JavaIndexInputMock`).
Not sure we can make it reusable. 

My view is that those `ON_CALL` maps to Spock's ` >> ` operator in Groovy, which is also heavily used in unit test method.

Reply:
Ah I think I saw the same file again and thought its copied, My bad ignore this one